### PR TITLE
[FLINK-34270][docs] Update doc for the new Table source & sink interfaces introduced in FLIP-146 & 367

### PR DESCRIPTION
**Pull request by X-czh**
## What is the purpose of the change

Update doc for the new Table source & sink interfaces introduced in FLIP-146 & 367.

## Brief change log

- Update doc for the new Table source & sink interfaces introduced in FLIP-146 & 367.
- Hotfix to mark `SinkV2Provider` as the recommend core interface for `SinkRuntimeProvider` instead of the deprecated `SinkProvider`

## Verifying this change

3fc0eac0-578f-404a-934c-18f5d125a5cd

42a6a968-a2fc-4069-9c6f-14a843ab202a

### CODE REVIEW

**Comment by X-czh** [comment]
@libenchao Could you help review the doc change?

File: docs/content/docs/dev/table/sourcesSinks.md [hunk 1]
```diff
@@ -197,10 +197,16 @@ A table source can implement further ability interfaces such as `SupportsProject
...
-The runtime implementation of a `ScanTableSource` must produce internal data structures. Thus, records
-must be emitted as `org.apache.flink.table.data.RowData`. The framework provides runtime converters such
+The returned _scan runtime provider_ provides the runtime implementation for reading the data. There are
+different interfaces for runtime implementation, among which `SourceProvider` is the recommended core interface.
+
+Independent of the provider interface, the source runtime implementation must produce internal data structures.
+Thus, records must be emitted as `org.apache.flink.table.data.RowData`. The framework provides runtime converters such
 that a source can still work on common data structures and perform a conversion at the end.
 
+To support parallelism setting, the dynamic table factory should support the optional `scan.parallelism` option
```
Comment (libenchao) [inline docs/content/docs/dev/table/sourcesSinks.md:207]: Is it worth to mention that `scan.parallelism` has been defined in `FactoryUtil`?

**Review by libenchao — COMMENTED** [review]
@X-czh Looks good, I just have one minor comment.

File: docs/content/docs/dev/table/sourcesSinks.md (hunk 1 above)
Comment (X-czh) [reply docs/content/docs/dev/table/sourcesSinks.md:207]: Updated to mention it, thanks~

**Comment by X-czh** [comment]
@flinkbot run azure

**Pull request closed without merging**